ResearchKit/Common/ORKPageStep.h

```diff
@@ -0,0 +1,132 @@
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import <ResearchKit/ORKStep.h>
+#import <ResearchKit/ORKOrderedTask.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ The `ORKPageStep` class is a concrete subclass of `ORKStep`, used for presenting a subgrouping of
+ `ORKStepViewController` views using a `UIPageViewController`.
+ 
+ To use `ORKPageStep`, instantiate the object, fill in its properties, and include it in a task. 
+ Next, create a task view controller for the task and present it.
+ 
+ The base class implementation will instatiate a read-only `ORKPageStepViewController` to display
+ a series of substeps. For each substep, the `ORKStepViewController` will be instantiated and added
+ as a child of the `UIPageViewController` contained by the parent `ORKPageStepViewController`..
+ 
+ Customization can be handled by overriding the base class implementations in either `ORKPageStep`
+ or `ORKPageStepViewController`.
+ */
+
+ORK_CLASS_AVAILABLE
+@interface ORKPageStep : ORKStep
+
+/**
+ Returns an initialized page step using the specified identifier and array of steps.
+ 
+ @param identifier  The unique identifier for the step.
+ @param steps       An array of `ORKStep` objects in the order in which they should be presented.
+ 
+ @return An initialized page step.
+ */
+- (instancetype)initWithIdentifier:(NSString *)identifier
+                             steps:(nullable NSArray<ORKStep *> *)steps NS_DESIGNATED_INITIALIZER;
+
+/**
+ Returns an initialized page step using the specified identifier and array of steps.
+ 
+ @param identifier  The unique identifier for the step.
+ @param task        The task used to run the subtask.
+ 
+ @return An initialized page step.
+ */
+- (instancetype)initWithIdentifier:(NSString *)identifier
+                          pageTask:(ORKOrderedTask *)task NS_DESIGNATED_INITIALIZER;
+
+/**
+ Returns a page step initialized from data in the given unarchiver.
+ 
+ A page step can be serialized and deserialized with `NSKeyedArchiver`. Note
+ that this serialization includes strings that might need to be localized.
+ 
+ @param aDecoder    The coder from which to initialize the ordered task.
+ 
+ @return An initialized ordered task.
+ */
+- (instancetype)initWithCoder:(NSCoder *)aDecoder NS_DESIGNATED_INITIALIZER;
+
+/**
+ The subtask used to determine the next/previous steps that are in this grouping
+ */
+@property (nonatomic, copy, readonly) ORKOrderedTask *pageTask;
```

Comment:
To simplify the encoding so that it would work for ORKTest serialization.  Initially, I was using `id <ORKTask, NSCopying, NSSecureCoding>` but I couldn't figure out how to get the ORKTest serialization tests to work correctly.